File: dsl-codegen/src/main/java/cbs/dsl/codegen/RegistrationSpec.java
package cbs.dsl.codegen;

import cbs.dsl.api.DslComponent.DslComponentModel;

public record RegistrationSpec(
    String packageName,
    String className,
    String code,
    DslInterfaceType interfaceType,
    String inputType,
    String outputType,
    DslComponentModel componentModel,
    String dslBody,
    String dslImports,
    String sourceCode) {

  public RegistrationSpec(
      String packageName,
      String className,
      String code,
      DslInterfaceType interfaceType,
      String inputType,
      String outputType,
      DslComponentModel componentModel,
      String dslBody,
      String dslImports) {
    this(packageName, className, code, interfaceType, inputType, outputType, componentModel, dslBody, dslImports, null);
  }

  public RegistrationSpec(
      String packageName,
      String className,
      String code,
      DslInterfaceType interfaceType,
      String inputType,
      String outputType,
      DslComponentModel componentModel) {
    this(packageName, className, code, interfaceType, inputType, outputType, componentModel, null, null, null);
  }
}

File: dsl-codegen/src/main/kotlin/cbs/dsl/codegen/RegistrationSpec.kt
package cbs.dsl.codegen

/**
 * Represents a DSL component to be registered in the [cbs.dsl.impl.ImplRegistry].
 *
 * @param packageName The package containing the annotated class.
 * @param className The simple name of the annotated class.
 * @param code The code identifier from the [cbs.dsl.api.DslComponent] annotation.
 * @param interfaceType The type of DSL definition interface implemented.
 */
data class RegistrationSpec(
    val packageName: String,
    val className: String,
    val code: String,
    val interfaceType: DslInterfaceType,
)

/** Enumerates the supported DSL definition interfaces that can be registered. */
enum class DslInterfaceType {
  TRANSACTION,
  HELPER,
  CONDITION,
  WORKFLOW,
  MASS_OPERATION,
}
